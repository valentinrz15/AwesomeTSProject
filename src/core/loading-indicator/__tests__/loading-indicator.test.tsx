import React from 'react';
import renderer from 'react-test-renderer';
import { LoadingIndicator } from '../index';

const LOADING_INDICATOR_COLOR = '#000000';

describe('loading indicator test', () => {
  it('should render loading indicator with size small', () => {
    const render = renderer.create(<LoadingIndicator size="small" />).toJSON();

    expect(render).toMatchSnapshot();
  });
  it('should render loading indicator with size large', () => {
    const render = renderer.create(<LoadingIndicator size="large" />).toJSON();
    expect(render).toMatchSnapshot();
  });
  it('should render loading indicator with color', () => {
    const render = renderer
      .create(<LoadingIndicator color={LOADING_INDICATOR_COLOR} />)
      .toJSON();

    expect(render).toMatchSnapshot();
  });
  it('should render loading indicator without color and size', () => {
    const render = renderer.create(<LoadingIndicator />).toJSON();

    expect(render).toMatchSnapshot();
  });
});
