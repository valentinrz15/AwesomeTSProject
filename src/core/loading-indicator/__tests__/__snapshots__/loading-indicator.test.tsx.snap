// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading indicator test should render loading indicator with color 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="container-loading-indicator"
>
  <ActivityIndicator
    color="#000000"
  />
</View>
`;

exports[`loading indicator test should render loading indicator with size large 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="container-loading-indicator"
>
  <ActivityIndicator
    size="large"
  />
</View>
`;

exports[`loading indicator test should render loading indicator with size small 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="container-loading-indicator"
>
  <ActivityIndicator
    size="small"
  />
</View>
`;

exports[`loading indicator test should render loading indicator without color and size 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="container-loading-indicator"
>
  <ActivityIndicator />
</View>
`;
